dist/kothic-leaflet-clickable.js

```diff
@@ -0,0 +1,70 @@
+L.TileLayer.Kothic.Clickable = L.TileLayer.Kothic.extend({
+    options: {
+        pixelThreshold:10
+    },
+
+    initialize: function(url, options) {
+        this._data = {};
+        L.TileLayer.Kothic.prototype.initialize.call(this, url, options); 
+    },
+
+    onAdd: function(map) {
+        this._map = map;
+        this._map.on("click", L.Util.bind(this._onclick, this));
+        L.TileLayer.Canvas.prototype.onAdd.call(this, map);
+    },
+
+    _onclick: function(e) {
+        var tileClickPos = this._map.project(e.latlng);
+        var tileX = Math.floor(tileClickPos.x / this.options.tileSize);
+        var tileY = Math.floor(tileClickPos.y / this.options.tileSize);
+        var key = [this._map.getZoom() - this.options.zoomOffset, tileX, tileY].join('/');
+        var pixelsToKothicUnits = this._granularity / this.options.tileSize;
+        tileClickPos.x %= this.options.tileSize; 
+        tileClickPos.y %= this.options.tileSize; 
+        var dataX = tileClickPos.x * pixelsToKothicUnits;
+        // we've already inverted the Y axis of the data
+        var dataY = tileClickPos.y * pixelsToKothicUnits;
+        var threshold = this.options.pixelThreshold * pixelsToKothicUnits;
+
+        var lowest = Number.MAX_VALUE, nearestFeature = null;
+        for(var i = 0; i < this._data[key].features.length; i++) {
+            var dx = Math.abs(this._data[key].features[i].coordinates[0] - dataX), dy = Math.abs(this._data[key].features[i].coordinates[1] - dataY);
```

Comment:
It looks like this one will not work on polygons? I don't see a safeguard on that.